Firestore/core/src/util/ordered_code.h

```diff
@@ -61,21 +61,29 @@ class OrderedCode {
   // compatibility.
 
   static void WriteString(std::string* dest, absl::string_view str);
-  static void WriteNumIncreasing(std::string* dest, uint64_t val);
-  static void WriteSignedNumIncreasing(std::string* dest, int64_t val);
-
-  /**
-   * Creates an encoding for the "infinite string", a value considered to
-   * be lexicographically after any real string.  Note that in the case of
-   * WriteInfinityDecreasing(), this would come before any real string as
-   * the ordering puts lexicographically greater values first.
-   */
+  static void WriteStringDecreasing(std::string* dest, absl::string_view str);
+  static void WriteNumIncreasing(std::string* dest, uint64_t num);
+  static void WriteNumDecreasing(std::string* dest, uint64_t num);
+  static void WriteSignedNumIncreasing(std::string* dest, int64_t num);
+  static void WriteSignedNumDecreasing(std::string* dest, int64_t num);
+
+  // WriteDoubleIncreasing() and WriteDoubleDecreasing()
+  // provide the same functionality for IEEE 754 doubles.
+  // Guarantees about floating point:
+  //   if (d1 < d2) then WDI(d1) < WDI(d2) and WDD(d1) > WDD(d2)
+  //   if (d1 > d2) then WDI(d1) > WDI(d2) and WDD(d1) < WDD(d2)
+  static void WriteDoubleIncreasing(std::string* dest, double num);
+  static void WriteDoubleDecreasing(std::string* dest, double num);
+
+  // Creates an encoding for the "infinite string", a value considered to
+  // be lexicographically after any real string.  Note that in the case of
+  // WriteInfinityDecreasing(), this would come before any real string as
+  // the ordering puts lexicographically greater values first.
   static void WriteInfinity(std::string* dest);
+  static void WriteInfinityDecreasing(std::string* dest);
 
-  /**
-   * Special string append that can only be used at the tail end of
-   * an encoded string -- blindly appends "str" to "*dest".
-   */
+  // Special string append that can only be used at the tail end of
+  // an encoded string -- blindly appends "str" to "*dest".
   static void WriteTrailingString(std::string* dest, absl::string_view str);
```

Comment:
Is there a reason to skip `WriteTrailingCord(std::string* dest, const absl::Cord& cord);` ?